vms/authentication/tests/test_login.py

```diff
@@ -45,79 +46,134 @@ def tearDownClass(cls):
-        '''
+        """
         authentication_page = self.authentication_page
-        self.authentication_page.server_url = self.live_server_url
-        authentication_page.login({
-            'username': 'volunteer',
-            'password': 'volunteer'
-        })
-        self.assertEqual(self.driver.current_url,
+        authentication_page.server_url = self.live_server_url
+        authentication_page.get_page(authentication_page.server_url, '')
+        username = password = 'volunteer'
+        self.login(username, password)
+
+        self.wait.until(
+            EC.presence_of_element_located(
+                (By.XPATH,
+                 "//h1[contains(text(), 'Volunteer Management System')]"))
+        )
+
+        self.assertEqual(authentication_page.remove_i18n(self.driver.current_url),
                          self.live_server_url + authentication_page.homepage)
 
-        with self.assertRaises(NoSuchElementException):
-            authentication_page.get_incorrect_login_message()
+        self.assertRaisesRegexp(NoSuchElementException,
+                                'Message: Unable to locate element: .alert-danger',
+                                authentication_page.get_incorrect_login_message)
         authentication_page.logout()
 
     def test_incorrect_volunteer_credentials(self):
-        '''
+        """
         Method to simulate logging in of a Invalid volunteer user and check if
         they are displayed an error and redirected to login page again.
-        '''
+        """
         authentication_page = self.authentication_page
-        self.authentication_page.server_url = self.live_server_url
-        authentication_page.login({
-            'username': 'volunteer',
-            'password': 'wrong_password'
-        })
+        authentication_page.server_url = self.live_server_url
+        authentication_page.get_page(authentication_page.server_url, '')
+        username = 'admin'
+        password = 'wrong_password'
+        self.login(username, password)
 
-        self.assertNotEqual(
-            self.driver.current_url,
-            self.live_server_url + authentication_page.homepage)
+        self.wait.until(
+            EC.presence_of_element_located(
+                (By.CSS_SELECTOR, '.' + AuthenticationPageLocators.INCORRECT_LOGIN_ERROR))
+        )
 
-        self.assertEqual(self.driver.current_url,
-                         self.live_server_url + authentication_page.url)
+        self.assertNotEqual(authentication_page.remove_i18n(self.driver.current_url),
+                            self.live_server_url + authentication_page.homepage)
 
-<<<<<<< ec806c6d047fd408e0fd890e7c6845e2f54622e6
+        self.assertEqual(authentication_page.remove_i18n(self.driver.current_url),
+                         self.live_server_url + authentication_page.url)
         self.assertNotEqual(authentication_page.get_incorrect_login_message(), None)
-"""
+
+    def test_login_page_after_authentication(self):
```

Comment:
I didn't get actually what you are testing in this function. Can you explain?

Reply:
Suppose I'm already logged in and try to access the login page again then we should be redirected to home page instead of login page.

Reply:
Thanks